### PR TITLE
[NPUW] Relaxed checks in Sliding Window patching to not fail Whisper

### PR DESCRIPTION
**Pull request by AsyaPronina**
### Details:
 - *Hot-fix after https://github.com/openvinotoolkit/openvino/pull/32891*
 - *Phi3 Sliding window patching is applied before attention mask is added to Whisper models. But as Whisper model doesn't need SWA patching, then added an early return if neither `attention_mask` nor `position_ids` were found*

### Tickets:
 - *N/A*

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp [hunk 1]
```diff
@@ -762,8 +765,6 @@ class Phi3SlidingMask : public ov::pass::ModelPass {
                 position_ids_node_ptr = i.get_node_shared_ptr();
             }
         }
-        OPENVINO_ASSERT(attention_mask_node_ptr, "attention_mask input is not found!");
```
Comment (AlexanderKalistratov) [inline src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:765]: Let's do `return` instead

Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:765]: made return!

**Comment by dmatveev** [comment]
This patch should never work on a whisper model first of all.

Second, are only older transformers impacted?
